vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1300,26 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private int getNumChildrenWithType(Node parent, Node child) throws RepositoryException {
```

Comment:
If there are many entries, counting can be slow. If all you need is "getNumChildrenWithType(child.getParent(), child) <= 1" then it's not needed to count the entries. What about "hasChildWithType(...)"?

Reply:
the problem is, that a node can only be deleted, if there are more than one child nodes with the respective type.
so this method needs to count. but as an optimization, the counting can be aborted when there are found 2.

Reply:
I can change that to two